store/store.go

```diff
@@ -413,6 +397,18 @@ func (s *Store[H]) flushLoop() {
 	for headers := range s.writes {
 		// add headers to the pending and ensure they are accessible
 		s.pending.Append(headers...)
+
+		if len(headers) > 0 && s.tailHeader.Load() == nil {
```

Comment:
Sadly `headers` can be `nil` here.

Reply:
nil there represents a closing signal

Reply:
true but still we need to skip this case via `len(headers) > 0`.

Reply:
yes, no objections to handling it